File: 1_week/cash.c
#include <stdio.h>
#include <math.h>

void main(void){
  float input;
  int coins = 0;;
  int cents;
  printf("Input floating number: ");
  scanf("%f",&input);

  cents = input * 100;
  printf("$%.2f is equal to %i cents.\n", input, cents);

  do {
    if (cents - 25>=0){
      cents -= 25;
      printf("Quarter added. Cents left: %i\n", cents);
      coins++;
    }
    else if(cents - 10>=0){
      printf("Dime added, Cents left: %i\n", cents);
      cents -= 10;
      coins++;
    }
    else if(cents - 5>=0){
      printf("Nickle added. Cents left: %i\n", cents);
      cents -= 5;
      coins++;
    }  
    else if(cents - 1>=0){
      printf("Penny added. Cents left: %i\n", cents);
      cents -= 1;
      coins++;
    }  
   }while (cents > 0);

  printf("Total coins: %i\n", coins);
}

File: 1_week/cash.go
package main

import "fmt"

func main(){
  var money float32
  var cents int
  var coins uint

  for(money <=0){
    fmt.Print("Enter an amount: ")
    fmt.Scan(&money)
  }

  cents = int(money*100);
  fmt.Printf("$%.2f is %d cents.\n", money, cents)
 
  for cents > 0 {
    if cents - 25 >= 0 {
      cents = cents - 25
      coins +=  1
      fmt.Printf("Quarter added. %d cents left.\n", cents)
    } else if cents - 10 >=0 {
      cents -= 10
      coins += 1
      fmt.Printf("Dime added. %d cents left.\n", cents)

    } else if cents -5 >= 0{
      cents -= 5
      coins += 1
      fmt.Printf("Nickel added. %d cents left. \n", cents)
    } else if cents -1 >=0 {
      cents -= 1
      coins += 1
      fmt.Printf("Penny added. %d cents left. \n", cents)
    }
  }

  fmt.Printf("Total coins: %d\n", coins)
}
